types/abi/src/recode_rlp.rs

```diff
@@ -207,6 +213,47 @@ impl Abi {
     }
 }
 
+pub fn rlp_encode(input: Vec<u8>) -> Vec<u8> {
+    let len = input.len();
+
+    let mut encoded: Vec<u8> = Vec::new();
+
+    // If the length is between 0 and 55 bytes and the first byte is greater than or equal to 0x80
+    if len > 0 && len <= 55 && input[0] >= 0x80 {
+        encoded.push(0x80 + len as u8);
+        encoded.extend(input);
+    }
+    // If the length is between 1 and 55 bytes and the first byte is less than 0x80
+    else if len > 0 && len <= 55 && input[0] < 0x80 {
+        encoded = input;
+    }
+    // If the length is more than 55 bytes
+    else if len > 55 {
+        let len_of_len = length_of_length(len);
+        encoded.push(0xb7 + len_of_len as u8);
+        encoded.extend(encode_length(len, len_of_len));
+        encoded.extend(input);
+    }
+
+    encoded
+}
+
+pub fn length_of_length(len: usize) -> usize {
+    let mut len_of_len = 0;
+    while (len >> (8 * len_of_len)) > 0 {
+        len_of_len += 1;
+    }
+    len_of_len
+}
+
+pub fn encode_length(len: usize, len_of_len: usize) -> Vec<u8> {
+    let mut encoded_length = vec![0; len_of_len];
+    for i in 0..len_of_len {
+        encoded_length[len_of_len - 1 - i] = ((len >> (8 * i)) & 0xff) as u8;
```

Comment:
I would suggest adding some notes on this statement for future peoples understanding, or even  a link to the spec where this comes from